includes/discovery/storage/datadomain.inc.php

```diff
@@ -22,6 +22,9 @@
         $total = $storage['fileSystemSpaceSize'] * $units;
         $used = $storage['fileSystemSpaceUsed'] * $units;
         if ($descr == "/data: post-comp") {
+            if (is_null($index)) {
+                $index=$tmp_index;
```

Comment:
$tmp_index is undefined...

Reply:
Why use fileSystemResourceIndex at all?

Reply:
I was torn on this as well. Older DataDomain versions do have the fileSystemResourceIndex on their MIB file where as modern DataDomains have completely removed it. 

I don't know if it served any purpose at all in the past. I agree with your change and I'll keep testing to see if it's used in different setups. Thanks.

Reply:
someone probably just didn't realize fileSystemResourceIndex  is in the table index.